pagefaultgames/pokerogue
# [Bug] Fix level 100 moves being skipped if leveling from < 100 to > 100

## What are the changes?
Currently, Pokemon that can learn moves at level 100 will not be prompted to learn them if they skip level 100 (e.g. by leveling straight from 99 to 101).

This PR fixes that bug.

## Why am I doing these changes?
it's a bug

## What did change?
The condition for triggering the `LearnMovePhase` was incorrectly written as `if (this.level <= 100)`, rather than `if (this.lastLevel < 100)`.

### Screenshots/Videos

https://github.com/user-attachments/assets/681c50c9-3e57-44db-9bba-20701149e562



## How to test the changes?
using overrides, probably
- level from 99 to 100 -> confirm you're prompted to learn the move
- level from 99 to >100 -> confirm you're still prompted to learn the move

## Checklist
